spark-submission-worker/src/main/java/org/apache/spark/k8s/operator/SparkAppResourceSpec.java

```diff
@@ -0,0 +1,127 @@
+import org.apache.spark.deploy.k8s.Constants;
+import org.apache.spark.deploy.k8s.KubernetesDriverSpec;
+import org.apache.spark.deploy.k8s.SparkPod;
+import org.apache.spark.deploy.k8s.submit.KubernetesClientUtils;
+
+/**
+ * Resembles resources that would be directly launched by operator. Based on resolved
+ * org.apache.spark.deploy.k8s.KubernetesDriverSpec, it:
+ *
+ * <ul>
+ *   <li>Add ConfigMap as a resource for driver
+ *   <li>Converts scala types to Java for easier reference from operator
+ * </ul>
+ *
+ * <p>This is not thread safe and not expected to be shared among reconciler threads
+ */
+public class SparkAppResourceSpec {
+  @Getter private final Pod configuredPod;
+  @Getter private final List<HasMetadata> driverPreResources;
+  @Getter private final List<HasMetadata> driverResources;
+  private final SparkAppDriverConf kubernetesDriverConf;
+
+  public SparkAppResourceSpec(
+      SparkAppDriverConf kubernetesDriverConf, KubernetesDriverSpec kubernetesDriverSpec) {
+    this.kubernetesDriverConf = kubernetesDriverConf;
+    String namespace = kubernetesDriverConf.sparkConf().get(Config.KUBERNETES_NAMESPACE().key());
+    Map<String, String> confFilesMap =
+        KubernetesClientUtils.buildSparkConfDirFilesMap(
+                kubernetesDriverConf.configMapNameDriver(),
+                kubernetesDriverConf.sparkConf(),
+                kubernetesDriverSpec.systemProperties())
+            .$plus(new Tuple2<>(Config.KUBERNETES_NAMESPACE().key(), namespace));
+    SparkPod sparkPod = addConfigMap(kubernetesDriverSpec.pod(), confFilesMap);
+    this.configuredPod =
+        new PodBuilder(sparkPod.pod())
+            .editSpec()
+            .addToContainers(sparkPod.container())
+            .endSpec()
+            .build();
+    this.driverPreResources =
+        new ArrayList<>(
+            JavaConverters.seqAsJavaList(kubernetesDriverSpec.driverPreKubernetesResources()));
+    this.driverResources =
+        new ArrayList<>(
+            JavaConverters.seqAsJavaList(kubernetesDriverSpec.driverKubernetesResources()));
+    this.driverResources.add(
+        KubernetesClientUtils.buildConfigMap(
+            kubernetesDriverConf.configMapNameDriver(), confFilesMap, new HashMap<>()));
+    // 'metadata.namespace' might be absent from template files.
+    this.driverPreResources.forEach(r -> setNamespaceIfMissing(r, namespace));
+    this.driverResources.forEach(r -> setNamespaceIfMissing(r, namespace));
+  }
+
+  private void setNamespaceIfMissing(HasMetadata resource, String namespace) {
+    if (StringUtils.isNotEmpty(resource.getMetadata().getNamespace())) {
+      return;
+    }
+    resource.getMetadata().setNamespace(namespace);
+  }
+
+  private SparkPod addConfigMap(SparkPod pod, Map<String, String> confFilesMap) {
+    Container containerWithVolume =
+        new ContainerBuilder(pod.container())
+            .addNewEnv()
+            .withName(org.apache.spark.deploy.k8s.Constants.ENV_SPARK_CONF_DIR())
```

Comment:
Do you need a full package name when we have `import` statement? Maybe, the following?
```java
- .withName(org.apache.spark.deploy.k8s.Constants.ENV_SPARK_CONF_DIR())
+ .withName(Constants.ENV_SPARK_CONF_DIR())
```